etc/compile/run_go_cmds.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+#
+# This file actually builds Pachyderm binaries ('pachd' and 'worker') and is
+# called by etc/compile/compile.sh inside the Pachyderm build container, but it
+# run as an unprivileged user with the host caller's user ID, instead of as
+# root. This prevents 'go build' from littering the pachyderm git directory
+# with root-owned files
+
+set -Eex
+
+env
+if [[ -n "${ROOT_PATH}" ]]; then
+  # Called from linux via 'su' -- must reset PATH
+  export PATH="${ROOT_PATH}"
+fi
+which go
+
+# Navigate to root of repo
+cd "$(dirname "${0}")/../.."
+
+BINARY="${1}"
+LD_FLAGS="${2}"
+PROFILE="${3}"
+
+mkdir -p _tmp
```

Comment:
would it make sense to use `mktemp -d` instead? maybe also instead of `docker_build_worker.tmpdir` and `docker_build_worker.tmpdir`?

Reply:
I removed the `mkdir -p _tmp` line, which I should've removed a while ago (we don't use `_tmp` for anything in the script). I originally changed the directory used for building the `pachd` and `worker` images from `_tmp` to `docker_build_xyz.tmpdir` just so that when the build failed, I wouldn't be stuck with this `_tmp` directory in my repo, trying to remember why it was there (i.e. the new names are supposed to be clearer).

IMO the advantage of using a static name rather than `mktemp` is that if we're trying to fix compilation errors, and running `make docker-build` repeatedly (and it's failing, due to the compilation errors that we're trying to fix), we don't accumulate `docker_build_xyz` directories as we go. Instead the `docker_build_xyz.tmpdir` directory is reused. But I really don't have a strong opinion about this—either approach seems completely fine to me. If you'd prefer I change it in light of my rationale I'm happy to.

Reply:
I think using `mktemp` makes more sense because on systems using tmpfs (including ubuntu and I imagine most linux distros):

1) The temporary directory is not persisted, so it's wiped out very regularly.
2) Because the contents are stored in-memory, it gives a big performance boost versus persisting to disk.

However, this isn't a blocker because really the long-term fix is multi-stage builds, which will obviate this anyways.

Reply:
This makes sense. I think on ubuntu, `/tmp` is actually a regular dir, while `/dev/shm` is a tmpfs mount. I couldn't actually build the docker image in `/dev/shm` because, by default, the ubuntu docker image limits that tmpfs mount to 64MB, which cause the build to fail with "out of space" errors.

But building in `/tmp` with regular `mktemp` worked, and, I realized, had the side benefit that we never leave temporary dirs on the host machine at all 🤦